src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioDocumentNavigationService.cs

```diff
@@ -251,7 +251,15 @@ static VsTextSpan GetVsTextSpan(SourceText text, int position, int virtualSpace)
             {
                 if (workspace.CurrentSolution.GetDocument(documentId) == null)
                 {
-                    var generatedDocument = workspace.CurrentSolution.GetProject(documentId.ProjectId)
+                    var project = workspace.CurrentSolution.GetProject(documentId.ProjectId);
+                    if (project is null)
+                    {
+                        // This is a source generated document shown in Solution Explorer, but is no longer valid since
+                        // the configuration and/or platform changed since the last generation completed.
+                        return false;
+                    }
+
+                    var generatedDocument = project
```

Comment:
Indenting in the lines later seems odd here.

Reply:
The indentation was also broken prior to this change, and was out of scope so I left it.